pkgs/build-support/rust/default.nix

```diff
@@ -35,6 +35,7 @@
 , cargoUpdateHook ? ""
 , cargoDepsHook ? ""
 , cargoBuildFlags ? []
+, cargoBuildFlagsMultiple ? []
```

Comment:
I agree with @symphorien . The `buildRustPackage` API is already quite complex as it is. Unless this use-case becomes very frequent in nixpkgs, I think it is better that projects that need to do multiple builds with different feature flags use their own `buildPhase`.

As @lopsided98 points out, such cases are very well handled by `buildRustCrate` plus `crate2nix`, since it compiles every crate as a separate output crate.

> I hadn't seen that, thanks. I would prefer if the mechanism lived in nixpkgs itself, so I'm suggesting that we do merge this PR. I'll also take a look at crate2nix nonetheless.

`crate2nix` is in nixpkgs and uses the `buildRustCrate` functionality in nixpkgs. If you do not want to add a `Cargo.nix` file in your repository and keep it up to date, `crate2nix` also supports IFD to generate the Nix from `Cargo.lock`.